src/libraries/System.Reflection/tests/TypeInfoTests.cs

```diff
@@ -1569,6 +1570,49 @@ public void IsSZArray(Type type, bool expected)
             Assert.Equal(expected, type.GetTypeInfo().IsSZArray);
         }
 
+        public static IEnumerable<object[]> TypeNamesToVerifyData()
+        {
+            yield return new object[] { "Some.Assembly.FakeType, Some.Assembly, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089", "Some.Assembly, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" }; // full
```

Comment:
This is a lot of tests for remote executor. Each of these lines will launch a new process. Do we really need to run this inside remote executor?

Reply:
I think we ought to because it registers the callback and we don't want that to pollute the normal environment? I could use an array with the strings and just run a loop inside a single remote executor thread, but I wanted to try and follow the normal xunit theory pattern. Happy to change it if you think that would be better or have some other suggestion.

Reply:
The test assembly name can be unique, the callback can filter out everything except the unique name as the first thing, and the test specific callback can be unregistered at the end of the test in finally block.

The pollution of the environment should be none/minimal with this. (The runtime may cache the failure to load the assembly with the unique name that should be acceptable.)

Reply:
Works for me, thanks for the suggestion. Will fix.

Reply:
Actually, if the runtime caches the failure this won't work, right? The handler needs to be called each time here.

I don't think in practice either runtime will cache `AppDomain.AssemblyResolve`, but I'm not sure I like relying on that here.